compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -676,6 +676,23 @@ impl<'hir> LoweringContext<'_, 'hir> {
         {
             self.lower_expr(body)
         } else {
+            let removal_span = |mut removal_span: Span| {
```

Comment:
Is this necessary? Can you use `find_ancestor_in_same_ctxt` instead of cooking it yourself?

Reply:
You should be able to do like... `pat_span.find_ancestor_in_same_ctxt(arm_span)`.

Reply:
Done (amended last commit).